### PR TITLE
More async

### PR DESCRIPTION
**Pull request by vincenthz**
convert a chunks of function to std::future with 0.1 compat for the network

### CODE REVIEW

File: jormungandr/Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ imhamt = { path = "../chain-deps/imhamt" }
 error-chain = "0.12"
 futures    = "0.1"
 futures03 = { package = "futures", version = "0.3.4", features = ["compat"] }
+futures-util = "0.3"
```
Comment (mzabaluev) [inline jormungandr/Cargo.toml]: There should not be a need to use this directly, it's reexported from `futures03`

File: jormungandr/src/blockchain/chain.rs [hunk 2]
```diff
@@ -332,15 +338,15 @@ impl Blockchain {
     pub fn get_ref(
         &self,
         header_hash: HeaderHash,
-    ) -> impl Future<Item = Option<Arc<Ref>>, Error = Error> {
+    ) -> impl Future01<Item = Option<Arc<Ref>>, Error = Error> {
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/chain.rs]: Hmm, so you did not convert all `Blockchain` methods at once...

File: jormungandr/src/blockchain/process.rs [hunk 3]
```diff
@@ -765,7 +701,7 @@ async fn check_and_apply_block(
     let fragment_ids = block.fragments().map(|f| f.id()).collect::<Vec<_>>();
     let applied_block = blockchain
         .apply_and_store_block(post_checked, block)
-        .compat()
+        //.compat()
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/process.rs]: Just remove the line.

File: jormungandr/src/network/bootstrap.rs [hunk 4]
```diff
@@ -173,44 +174,68 @@ where
...
+                logger,
+                "validated block";
+                "hash" => %post_checked.header().hash(),
+                "block_date" => %post_checked.header().block_date(),
+            );
+            let applied = blockchain
+                .apply_and_store_block(post_checked, block)
+                .await
+                .map_err(|e| Error::ApplyBlockFailed { source: e })?;
+            Ok(applied.cached_ref()) //.map(|applied| applied.cached_ref())
```
Comment (mzabaluev) [inline jormungandr/src/network/bootstrap.rs]: Could clean up the commented code.

File: jormungandr/src/network/bootstrap.rs [hunk 5]
```diff
@@ -173,44 +174,68 @@ where
...
 }
 
-fn handle_block(
+fn process_new_ref_old(
+    logger: Logger,
+    blockchain: Blockchain,
+    tip: Tip,
+    candidate: Arc<Ref>,
+) -> impl Future<Item = (), Error = blockchain::Error> {
+    use futures_util::compat::Compat;
```
Comment (mzabaluev) [inline jormungandr/src/network/bootstrap.rs]: This is available directly from `futures03::compat`.

File: jormungandr/Cargo.toml (hunk 1 above)
Comment (vincenthz) [reply jormungandr/Cargo.toml]: thanks, it took me a while to find in the first place in future_util

File: jormungandr/src/blockchain/chain.rs (hunk 2 above)
Comment (vincenthz) [reply jormungandr/src/blockchain/chain.rs]: too many functions to change at once, this makes a lot of churn already, better to spread it along

**Pull request merged**